gaphor/ui/css_function_completion_provider.py

```diff
@@ -0,0 +1,100 @@
+from gi.repository import Gio, GObject, Gtk, GtkSource
+
+
+class CssFunctionCompletionProvider(GObject.GObject, GtkSource.CompletionProvider):
+    FUNCTIONS = [
+        "attr",
+        "calc",
+        "conic-gradient",
+        "counter",
+        "cubic-bezier",
+        "hsl",
+        "hsla",
+        "linear-gradient",
+        "max",
+        "min",
+        "radial-gradient",
+        "repeating-conic-gradient",
+        "repeating-linear-gradient",
+        "repeating-radial-gradient",
+        "rgb",
+        "rgba",
+        "var",
```

Comment:
I think we should limit this to the functions that Gaphor supports: `hsl`, `hsla`, `rgb`, and `rgba`.

Reply:
Well, I would guess anything that WebKit support would work?
I think the other functions would probably be possible to use too, even though I think they're unlikely to be much used, as many are for advanced uses and seldom used.

Reply:
> Well, I would guess anything that WebKit support would work? 

We're not using webkit for rendering. Instead we use Cairo and a the items render themselves on the canvas, with some styling.

Reply:
Oh, I am aware which CSS properties and functions can be used in Cairo.